backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -296,6 +301,71 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
     }
   }
 
+  private void deleteRoleBindingsOfAccessibleResources(List<RepositoryEntity> allowedResources) {
+    UserInfo unsignedUser = authService.getUnsignedUser();
+    for (RepositoryEntity repositoryEntity : allowedResources) {
+      String repositoryId = String.valueOf(repositoryEntity.getId());
+      String ownerRoleBindingName =
+          roleService.buildRoleBindingName(
+              ModelDBConstants.ROLE_REPOSITORY_OWNER,
+              repositoryId,
+              repositoryEntity.getOwner(),
+              ModelDBServiceResourceTypes.REPOSITORY.name());
+      RoleBinding roleBinding = roleService.getRoleBindingByName(ownerRoleBindingName);
+      if (roleBinding != null && !roleBinding.getId().isEmpty()) {
+        roleService.deleteRoleBinding(roleBinding.getId());
+      }
+
+      // Remove all repositoryEntity collaborators
+      roleService.removeResourceRoleBindings(
+          repositoryId, repositoryEntity.getOwner(), ModelDBServiceResourceTypes.REPOSITORY);
+
+      // Delete workspace based roleBindings
+      deleteWorkspaceRoleBindings(
+          repositoryEntity.getWorkspace_id(),
+          WorkspaceType.forNumber(repositoryEntity.getWorkspace_type()),
+          String.valueOf(repositoryEntity.getId()));
+    }
+  }
+
+  private void deleteWorkspaceRoleBindings(
+      String workspaceId,
+      WorkspaceType workspaceType,
+      String resourceId) {
+    if (workspaceId != null && !workspaceId.isEmpty()) {
```

Comment:
could we refactor the common logic to a common function which takes in the resourceType to delete as  parameters.
we have the deleteWorkspaceRoleBindings at project, dataset and repository now.

This function should exist and just invoke the common function.

Reply:
fixed